include/aws/common/assert.h

```diff
@@ -107,6 +107,15 @@ AWS_EXTERN_C_END
 #define __CPROVER_r_ok(base, len) (AWS_MEM_IS_READABLE(base, len))
 #define __CPROVER_w_ok(base, len) (AWS_MEM_IS_WRITEABLE(base, len))
 
+/* Logical consequence. */
+#define AWS_IMPLIES(a, b) (!(a) || (b))
+
+/**
+ * If and only if (iff) is a biconditional logical connective between statements a and b.
+ * Equivalent to (AWS_IMPLIES(a, b) && AWS_IMPLIES(b, a)).
+ */
+#define AWS_IFF(a, b) (!!(a) == !!(b))
```

Comment:
At first glance, it's not clear why we need double negations `!!` here since they _should_ simply cancel each other out (in logic). But you need it here for `AWS_IFF` to work correctly for non-Boolean `a` and `b` values. I think this should be documented, so someone doesn't simply optimizie it to `a == b` in future.

Also, we don't seem to be using `AWS_IFF` within this PR.

Reply:
+1

Reply:
I have implemented as suggested.